### DISCUSSION TITLE
Deleting user's traces via API to be GDPR compliant

### DISCUSSION THREAD

**Discussion by Stormyy** [langfuse/langfuse#2704] [Ideas]
### Describe the feature or potential improvement

We can already add an user id to a trace but we cannot yet delete any traces using this id via the API. This is necessary to be fully compliant as we need to delete traces whenever the user requests their account to be deleted according to GDPR

**Comment by marcklingen** [comment]
Thanks for raising this @Stormyy, we plan to add more data retention management tooling soon.

As GDPR compliance is crucial for many users of langfuse, you can:

- Langfuse Cloud: reach out to privacy@langfuse.com for data deletion requests, see https://langfuse.com/docs/data-security-privacy
- Self-hosted instance: delete traces from `traces` table based on `userId`. You will also need to delete corresponding `observations` and `scores`.

**Reply by EBazarov** [reply]
I will second what was been said above, looks as an important feature to have, especially when DB changes after release, before were PostgreSQL and now ClickHouse. Dropping traces and corresponding information via SQL request is not a huge deal, but it will be safer and more in complaince with GDPR if it will be possible to do via API.

**Comment by michaelgrewal** [comment]
I would like to be able to delete traces, spans, generations etc from the JS TS SDK

**Comment by marcklingen** [comment]
Forgot to update this thread, this is released: https://langfuse.com/docs/data-deletion

**Reply by clemra** [reply]
nice!
